### PR TITLE
Modularize CAP pipeline

### PR DESCRIPTION
**Pull request by HartS**
Changes in this PR:
- Break down jobs into component files
- Collapse repitition in many places
- Rename option to avail (HA vs SA)
- Integrate several newer merges to master

### CODE REVIEW

File: cap-ci/cap-pre-release.yaml.tmpl [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
-  deploy: true
+  deploy-k8s: false
+  deploy-kubecf: true
   smoke-tests: true
-  cf-acceptance-tests-brain: true
-  cf-acceptance-tests: true
-  sync-integration-tests: true
-  upgrade: true
-  stratos: true
+  cf-acceptance-tests-brain: false
```
Comment (prabalsharma) [inline cap-ci/cap-pre-release.yaml.tmpl]: should be true by default same for other jobs.

File: cap-ci/scripts/destroy_k8s.tmpl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{{ define "scripts_destroy_k8s" }}
+# delete k8s cluster
```
Comment (prabalsharma) [inline cap-ci/scripts/destroy_k8s.tmpl]: I guess you are using `{{- /*  */ }}` for comments

File: cap-ci/cap-pre-release.yaml.tmpl [hunk 3]
```diff
@@ -1,25 +1,37 @@
-create-k8s: false
-s3bucket: cap-release-artifacts
-
+{{ define "config" }}
 workertags:
   # caasp4: suse-internal
 
 # Currently only accepts disabling jobs if all later jobs are also disabled.
```
Comment (bikramnehra) [inline cap-ci/cap-pre-release.yaml.tmpl]: Is this still true? I thought this PR will address dependency issues.

File: cap-ci/cap-pre-release.yaml.tmpl [hunk 4]
```diff
@@ -38,3 +50,8 @@ cats:
   nodes: 3
   flake-attempts: 5
   timeout-scale: "3.0"
+s3:
```
Comment (bikramnehra) [inline cap-ci/cap-pre-release.yaml.tmpl:51]: This should be `CAP` bundle and not `kubecf`. 

```
bucket: cap-release-artifacts
regexp: CAP-(.*).tgz
```

**Review by bikramnehra — CHANGES_REQUESTED** [review]
Its really hard to go through all the files to check for consistency, however overall the structure looks good. Hopefully one full test case has been run using this pipeline structure.

File: cap-ci/cap-pre-release.yaml.tmpl (hunk 3 above)
Comment (HartS) [reply cap-ci/cap-pre-release.yaml.tmpl]: Good catch, thanks!

File: cap-ci/cap-pre-release.yaml.tmpl (hunk 4 above)
Comment (HartS) [reply cap-ci/cap-pre-release.yaml.tmpl:51]: @bikramnehra  this is consistent with master right now
- https://github.com/SUSE/cap/blob/a55a04a1133a99a5de0c3dac9b13e52ed60fc584/cap-ci/cap-pre-release.yaml#L2
- https://github.com/SUSE/cap/blob/a55a04a1133a99a5de0c3dac9b13e52ed60fc584/cap-ci/pipeline.template#L63

cap-release does indeed use the `CAP-(.*).tgz` regexp

**Review by bikramnehra — APPROVED** [review]
LGTM.

**Pull request merged**